src/Lecture1.hs

```diff
@@ -108,8 +121,23 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
-
+split :: Char -> String -> String -> [String]
+split c [] [] = []
+split c acc [] = [acc]
+split c acc (x:xs)
+    | (x == c) && (not $ null acc) = [acc] ++ (split c "" xs)
+    | x == c = split c acc xs
+    | otherwise = split c (acc ++ [x]) xs
+
+sumStr :: [String] -> Int
+sumStr [] = 0
+sumStr (x:xs) = (read x :: Int) + sumStr xs
+
+strSum :: String -> Int
+strSum "" = 0
+strSum str = sumStr $ split ' ' "" str
```

Comment:
You can use standard functions to simplify the implementation of `strSum`:

* `words`: to split a string into a list of space-separated words
* `read`: to parse an `Int` from `String`
* `sum`: to find a sum of all numbers in the list